#!/usr/bin/env python
# SLAW OBJECT DETECTION

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t, str_t
#from local_planner_limits import add_generic_localplanner_params

gen = ParameterGenerator()

gen.add("save", bool_t, 0, "height of camera", False)

gen.add("hough_dp",     double_t, 0, "", 2.0, 1.0, 20.0)
gen.add("hough_rmax",   int_t,    0, "", 0, 0, 640)
gen.add("hough_rmin",   int_t,    0, "", 0, 0, 640)
gen.add("hough_thresh", double_t, 0, "", 300, 0, 2000)
gen.add("hough_dmin",   double_t, 0, "", 64, 1, 640)
gen.add("canny_high",   double_t, 0, "", 100, 1, 2000)
gen.add("mask_x",       int_t,    0, "", 150, 0, 640)
gen.add("mask_y",       int_t,    0, "", 150, 0, 480)
gen.add("mask_width",   int_t,    0, "", 250, 0, 640)
gen.add("mask_height",  int_t,    0, "", 250, 0, 480)

exit(gen.generate("slaw_object_detection", "HoughCircles", "HoughCircles"))
